monai/bundle/meta_schema.json

```diff
@@ -0,0 +1,218 @@
+            "type": "string"
+        },
+        "numpy_version": {
+            "description": "Version of NumPy the bundlewas generated with.",
+            "type": "string"
+        },
+        "required_packages_version": {
+            "description": "Dictionary relating required additional package names to their versions.",
+            "type": "object"
+        },
+        "optional_packages_version": {
+            "description": "Dictionary relating optional additional package names to their versions.",
+            "type": "object"
+        },
+        "task": {
+            "description": "Plain-language description of what the bundle is meant to do.",
+            "type": "string"
+        },
+        "description": {
+            "description": "Longer form description of what the bundle is, what it does, etc.",
+            "type": "string"
+        },
+        "authors": {
+            "description": "State author(s) of the bundle.",
+            "type": "string"
+        },
+        "copyright": {
+            "description": "State copyright of the bundle.",
+            "type": "string"
+        },
+        "data_source": {
+            "description": "Where to download or prepare the data used in this bundle.",
+            "type": "string"
+        },
+        "data_type": {
+            "description": "Type of the data, like: `dicom`, `nibabel`, etc.",
+            "type": "string"
+        },
+        "image_classes": {
+            "description": "Description for every class of the input tensors.",
+            "type": "string"
+        },
+        "label_classes": {
+            "description": "Description for every class of the input tensors if present.",
+            "type": "string"
+        },
+        "pred_classes": {
+            "description": "Description for every class of the output prediction(s).",
+            "type": "string"
+        },
+        "eval_metrics": {
+            "description": "Dictionary relating evaluation metrics to the achieved scores.",
+            "type": "object"
+        },
+        "intended_use": {
+            "description": "What the bundle is to be used for, ie. what task it accomplishes.",
+            "type": "string"
+        },
+        "references": {
+            "description": "List of published referenced relating to the bundle.",
+            "type": "array"
+        },
+        "network_data_format": {
+            "$ref": "#/$defs/network_io"
+        }
+    },
+    "$comment": "This permits definitions for multiple networks with format <network-name>_data_format",
+    "patternProperties": {
+        "^[_a-zA-Z0-9]+_data_format$": {
+            "$ref": "#/$defs/network_io"
+        }
+    },
+    "required": [
+        "schema",
+        "version",
+        "monai_version",
+        "pytorch_version",
+        "numpy_version",
+        "required_packages_version",
+        "optional_packages_version",
```

Comment:
Since we already know all of the packages we need after the bundle was created, we could remove 'optional_packages_version', which may be confusing, and instead, directly use 'required_packages_version'. What do you think? 
cc @Nic-Ma @yiheng-wang-nv

Reply:
My idea was that `required_packages_version` is for additional packages this bundle absolutely required which wouldn't be present with a basic MONAI install, eg. `nibabel` which would be present only if the option was specified. `optional_packages_version` are ones which the bundle uses but doesn't strictly need.

Reply:
Yes, I know your points. But after bundle created, the creator needs to determine which packages are optional and then finished metadata.json. For users, even if the user knows that a package is optional, and they want to use the cheapest version, they still need to modify the config file themselves, so does an optional package still make sense for the bundle? Just wondering if it's necessary to complicate this, I'm open to any opinions.

Reply:
It's not necessary that the creator or the user has to modify anything. We have networks, transforms, and other components which check optional dependencies and will work fine if one isn't present. As an example, I'm working on a bundle which, in code in the `scripts` directory, optionally uses cupy but will resort to numpy if that isn't present.

Reply:
We may need to clearly describe what the "optional package" means here to avoid confusion. I was also confused at the first look..

Thanks.

Reply:
I've updated the schema description line to be a bit clearer. We would also document in docs and wherever else relevant that the required packages are absolutely needed by the bundle, but the optional packages enable features or capabilities not strictly needed but may allow broader behaviours or faster performance. Eg. if an operation somewhere can use cupy to accelerate something but will default to numpy if this isn't present then cupy would be an optional package, or nibabel would be optional if the bundle can work with just numpy array loading and not need nifti loading.